### ISSUE TITLE
Bad SQL generated when setting data type and nullability in PostgreSQL's ALTER TABLE .. ALTER .. SET statement

### ISSUE THREAD

**Issue by Eli8838** [jOOQ/jOOQ#8402]
### Expected behavior and actual behavior:
when running the following code for changing the size of the column varchar:

```
dslContext.alterTable(tableName)
                    .alterColumn(columnName)
                    .set(dataType.length(size))
                    .execute();
```

i get an error "Unexpected end of function definition at end of input Position"  since there is a missing semicolon before the 'end' in the statement generated:

`do $$ begin alter table "Users" alter "name" type varchar(255); alter table "Users" alter "name" set not null end $$`
### Versions:

- jOOQ: 3.11.5
- Java: 11
- Database (include vendor): Postgres 11.1
- OS: macOs

**Comment by lukaseder** [comment]
Thank you very much for your report. Indeed, I can reproduce this. It only happens when the data type passed to `set()` has an explicit nullability set. It seems, this regression was introduced with #3805

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.12. Backport scheduled to 3.11.11 (#8445)
